lib/auth.js

```diff
@@ -0,0 +1,148 @@
+/* global XMLHttpRequest */
+
+var defaultConfig = require('./config-default')
+
+/**
+ * Returns whether or not the current JS engine is in a browser environment. The
+ * environment is assumed to be Node.js if this returns false. Heuristically
+ * just looks for 'window' and 'XMLHttpRequest'.
+ * @returns {Boolean} true if the environment is a browser, false otherwise
+ */
+function isBrowser () {
+  return typeof window !== 'undefined' && typeof XMLHttpRequest !== 'undefined'
+}
+
+/**
+ * Enumerates the two main types of authentication endpoints.
+ * @readonly
+ * @enum {number}
+ */
+const AUTH_ENDPOINTS = {
+  PRIMARY: 0,
+  SECONDARY: 1
```

Comment:
Aren't these supposed to be strings and not ints? Sorry, I know I'm nitpicking. :)